### ISSUE TITLE
Memory usage

### ISSUE THREAD

**Issue by lnguyen** [philipperemy/deep-speaker#70]
Is there a way not to use so much memory if you add speakers?

**Comment by philipperemy** [comment]
You mean for training? I know the memory usage is not optimized for the softmax pre-training part (Somehow Keras needs double of the memory to work...). But the triplet training should be efficient in memory.

**Comment by lnguyen** [comment]
got it, thanks!
